### DISCUSSION TITLE
How to address the issue of a particular GitHub Integration being not reachable

### DISCUSSION THREAD

**Discussion by akmalick** [community/community#24547] [Apps, API and Webhooks]
<p>What happens in general if a GitHub integration is down.<br>
Suppose we have a GitHub App app that validates the pull request. What if that app is down, does GitHub display some message about it.</p>
<p>The pull request seems to go through when the app is not up.<br>
This is quite dangerous. I mean, is there any fallback mechanism that GitHub provides for GitHub integrations.</p>
<p><strong>EDIT 1 : START</strong></p>
<p>I was just wondering that since webhooks are sent from GitHub side to the webhook url that we configure in the app setting, is there a way for GitHub to notify users if the webhook url is not reachable.<br>
<strong>EDIT 1 : END</strong></p>

**Comment by jamesmartin** [comment, marked as answer]
<p>:wave: hi, <a class="mention" href="/akmalick">@akmalick</a>. Thanks for the question.</p>
<blockquote>
<p><a class="mention" href="/akmalick">@akmalick</a> wrote:</p>
</blockquote>
<blockquote>
<p>What happens in general if a GitHub integration is down.<br>
Suppose we have a GitHub App app that validates the pull request. What if that app is down, does GitHub display some message about it.</p>
</blockquote>
<blockquote>
<p>The pull request seems to go through when the app is not up.<br>
This is quite dangerous. I mean, is there any fallback mechanism that GitHub provides for GitHub integrations.</p>
</blockquote>
<p>GitHub only communicates with your App via <a href="https://developer.github.com/webhooks" rel="nofollow noopener">Webhooks, </a>sending HTTP requests to a URL of your choice. As the owner of the App you can access a <a href="https://developer.github.com/webhooks/testing/#listing-recent-deliveries" rel="nofollow noopener">list of recent Webhook deliveries</a> and see the payload we sent. If your Webhook URL does not respond with a 200, then the delivery is marked as failed. You have the option to “Redeliver” any Webhook, which will send the request again.</p>
<p>This information is <em>only</em> visible to the owner of the App. GitHub never displays this information (or anything else about whether your App is functioning correctly) to people who have installed your App.</p>
<p>Depending on the volume of traffic your App receives, it might be sufficient to manually “Redeliver” any failed Webhooks that your App missed during its downtime (as described above). If it is important that your App never misses an event, you could build a recovery mechanism, using the GitHub API to fetch any data that they may have missed during your App’s downtime.</p>
<p>Hope this helps!</p>
